libs/datamodel/connectors/datamodel-connector/src/lib.rs

```diff
@@ -95,6 +97,32 @@ pub trait Connector: Send + Sync {
         self.referential_actions(integrity).contains(action)
     }
 
+    /// This is used by the query engine schema builder.
+    ///
+    /// For a given scalar type + native type combination, this method should return the name to be
+    /// given to the filter input objects for the type. The significance of that name is that the
+    /// resulting input objects will be cached by name, so for a given filter input object name,
+    /// the filters should always be identical.
+    fn scalar_filter_name(&self, scalar_type_name: String, _native_type_name: Option<&str>) -> Cow<'_, str> {
```

Comment:
This can become more typed once we have more context in schema-builder (we have a path towards making that happen). Concretely, the signature would turn into something like `fn scalar_filter_name(&self, scalar_type: ScalarType, native_type: NativeTypeId) -> &'static str`.